docs/development/test-install.md

```diff
@@ -0,0 +1,133 @@
+checkout, so we build a tarball of one of those first:
+
+```
+./tools/build-release-tarball test-installer
+```
+
+This will produce a file in /tmp, which it will print out the path to
+as the last step; for example,
+`/tmp/tmp.fepqqNBWxp/zulip-server-test-installer.tar.gz`
+
+Next, unpack that file into a local directory; we will make any
+changes we want in our source checkout and copy them into this
+directory. The test installer needs the release directory to be named
+`zulip-server`, so we rename it and move it appropriately.  In the
+first line, you'll need to substitute the actual path that you got for
+the tarball, above:
+
+```
+tar xzf /tmp/tmp.fepqqNBWxp/zulip-server-test-installer.tar.gz
+mkdir zulip-test-installer
+mv zulip-server-test-installer zulip-test-installer/zulip-server
+```
+
+You should delete and re-create this `zulip-test-installer` directory
+(using these steps) if you are working on a different installer
+branch, or a significant time has passed since you last used it.
+
+### Test an install
+
+The `test-install` tooling takes a distribution release name
+(e.g. "focal" or "bionic"), the path to an unpacked release directory
+or tarball, and then any of the normal options you want to pass down
+into the installer.
+
+For example, to test an install onto Ubuntu 20.04 "Focal", we might
+call:
+
+```
+sudo ./tools/test-install/install \
+  -r focal \
+  ./zulip-test-installer/ \
+  --hostname=zulip.example.net \
+  --email=username@example.net
+```
+
+The first time you run this command for a given distribution, it will
+build a "base" image for that to use on subsequent times; this will
+take a while.
+
+### See running containers after installation
+
+Regardless of if the install succeeds or fails, it will stay running
+so you can inspect it. You can see all of the containers which are
+running, and their randomly-generated names, by running:
+
+```
+sudo lxc-ls -f
+```
+
+### Connect to a running container
+
+After using `lxc-ls` to list containers, you can choose one of them
+and connect to its terminal:
+
+```
+sudo lxc-attach --clear-env -n zulip-install-focal-PUvff
+```
+
+### Stopping and destroying containers
+
+When you've finished using a container, you should stop and destroy
+it:
+
+```
+sudo lxc-destroy -f -n zulip-install-focal-PUvff
+```
+
+### Iterating on the installer
+
+Iterate on the installer by making changes to your source tree,
+copying them into the release directory, and re-running the installer,
+which will start up a new container. Here, we update just the
+`scripts` and `puppet` directories of the release directory:
+
+```
+rsync -az scripts puppet zulip-test-installer/zulip-server/
+
+sudo ./tools/test-install/install \
+ -r focal \
+ ./zulip-test-installer/ \
+ --hostname=zulip.example.net \
+ --email=username@example.net
+```
+
+### Stopping and destroying all containers
+
+```
+sudo ./tools/test-install/destroy-all -f
+```
```

Comment:
This won't destroy the base image, only the test containers on top of it.

That's probably worth mentioning explicitly, both (a) to reassure people that it's not going to make them rebuild the base image next time and (b) so people know if they *do* want to destroy the base image (like because they think they're done for a while, and need the disk space), they'll need something else.